site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -0,0 +1,248 @@
+            return "publication-technicalnote"
+        elif document_type == "proceedings":
+            return "publication-conferenceproceeding"
+        elif document_type == "report":
+            return "publication-report"
+        elif document_type == "thesis":
+            return "publication-thesis"
+        else:
+            self.metadata_errors.append(
+                f"Error occurred while mapping document_type to resource_type. Couldn't fine a mapping rule for "
+                f"document_type {document_type}. INSPIRE record id: {self.inspire_metadata.get('control_number')}."
+            )
+            return None
+
+    def _transform_creators(self):
+        """Mapping of INSPIRE authors to creators and contributors."""
+        creators = []
+        authors = self.inspire_metadata.get("authors")
+        try:
+            for author in authors:
+                creators.append(
+                    {
+                        "person_or_org": {
+                            "type": "personal",
+                            "family_name": author.get("full_name"),
+                        }
+                    }
+                )
+            return creators
+        except Exception as e:
+            self.metadata_errors.append(
+                f"Error occurred while mapping INSPIRE authors to creators and contributors. INSPIRE record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
+            )
+            return None
+
+    def _transform_alternate_identifiers(self):
+        """Mapping of alternate identifiers."""
+        identifiers = []
+        inspire_id = self.inspire_metadata.get("control_number")
+        try:
+            # add INSPIRE id
+            identifiers.append({"identifier": str(inspire_id), "scheme": "inspire"})
+            return identifiers
+        except Exception as e:
+            self.metadata_errors.append(
+                f"Error occurred while mapping alternate identifiers. INSPIRE record id: {inspire_id}. Error: {e}."
+            )
+            return None
+
+    def transform_metadata(self):
+        """Transform INSPIRE metadata."""
+        rdm_metadata = {}
+
+        title, additional_titles = self._transform_titles()
+        rdm_metadata["title"] = title
+        if additional_titles:
+            rdm_metadata["additional_titles"] = additional_titles
+
+        # publication_date Note: tbd with Library which field we should use
+        publication_date = self._transform_publication_date()
+        rdm_metadata["publication_date"] = publication_date
+
+        # resource_type
+        resource_type = self._transform_document_type()
+        rdm_metadata["resource_type"] = {"id": resource_type}
+
+        # creators
+        creators = self._transform_creators()
+        rdm_metadata["creators"] = creators
+
+        # alternate identifiers
+        identifiers = self._transform_alternate_identifiers()
+        rdm_metadata["identifiers"] = identifiers
```

Comment:
to improve readability
```suggestion
        title, additional_titles = self._transform_titles()
        return {
             "title": title,
             "additional_titles": additional_titles,
             "publication_date" : self._transform_publication_date(),
             "resource_type": self._transform_document_type(),
             "creators": self._transform_creators(),
             "identifiers": self._transform_alternate_identifiers()
         }
      
```